packages/grid/src/GridRenderer.ts

```diff
@@ -3053,6 +3050,155 @@ export class GridRenderer {
 
     context.translate(-barLeft, -barTop);
   }
+
+  /**
+   * Gets the token boxes that are visible in the cell
+   * @param column The visible column
+   * @param row The visible row
+   * @param state The GridRenderState
+   * @returns An array of TokenBox of visible tokens or empty array
+   */
+  getTokenBoxesForVisibleCell(
+    column: VisibleIndex,
+    row: VisibleIndex,
+    state: GridRenderState
+  ): TokenBox[] {
+    return this.getCachedTokenBoxesForVisibleCell(column, row, state);
+  }
+
+  getCachedTokenBoxesForVisibleCell = memoizeClear(
+    (
+      column: VisibleIndex,
+      row: VisibleIndex,
+      state: GridRenderState
```

Comment:
This memoization right now is kind of useless - the `state` object is changed every time the mouse moves. What we _could_ memoize is the position of token boxes relative to the cell; that would require just the text, tokens for the cell, font, and text width as inputs. Would then need to translate in the `getTokenBoxesForVisibleCell` method, e.g.
```
getTokenBoxesForVisibleCell(column, row, state) {
  ...
  const text = model.textForCell(modelColumn, modelRow);
  const { width: textWidth, x: textX, y: textY } = this.getTextRenderMetrics(state, column, row);
  ... get truncated text
  const tokens = model.tokensForCell(modelColumn, modelRow, truncatedText.length);
  return this.getCachedTokenBoxes(text, tokens, font, textWidth, context).map(tokenBox => ({
    x1: tokenBox.x1 + textX,
    y1: tokenBox.y1 + textY,
    ...
  });
}

getCachedTokenBoxesForVisibleCell = memoizeClear((text, tokens, 
```